.github/workflows/release.yml

```diff
@@ -0,0 +1,46 @@
+Name: Publish Zarf Packages on Tag
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - 'v*'
+
+jobs:
+  push-resources:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install GoLang
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16.x
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Build Things For Release
+        run: make build-cli init-package
+      - name: Set AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_GOV_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_GOV_SECRET_ACCESS_KEY }}
+          aws-region: us-gov-west-1
+      - name: Push Release Artifacts to S3 Bucket
+        run: aws s3 cp build s3://zarf-public/release/${{ github.ref_name }} --region us-gov-west-1 --recursive
```

Comment:
@jeff-mccoy Do we need to be uploading release artifacts to an S3 bucket if we can just include them as assets in the GitHub release as we're doing below?

Reply:
I personally see value in having the files in s3 and on GitHub, would give our govt customers confidence that zarf releases were accessible even if there are issues with GitHub.  Also makes it easy to script version updates as the url is predictable--not sure if GitHub release artifact urls are, I know gitlab's we're not.

Reply:
That being said, if we wanted to just have this cut the release on GitHub and create artifacts without s3 for this issue, we can make a new issue for s3 later on and discuss there.

Reply:
Github’s URLs are definitely predictable, and as an added bonus there is even a URL to get artifacts from the “latest” release.

Reply:
Keeping release artifacts on s3 as a backup incase GitHub goes down at an inopportune time is simple and reasonable. 

I'll just need someone to add the `AWS_GOV_ACCESS_KEY_ID` and `AWS_GOV_SECRET_ACCESS_KEY` that we were using in the GitLab pipeline to the GitHub actions secrets.

Reply:
I added the new secrets